nilearn/_utils/niimg_conversions.py

```diff
@@ -134,8 +161,11 @@ def concat_niimgs(niimgs, dtype=np.float32, accept_4d=False,
         A single image.
     """
 
+    if len(niimgs) == 0:
+        raise TypeError('Cannot concatenate empty objects')
```

Comment:
What if niimgs is an iterator (which seems to be a possibility if you trust the docstring)?

IMHO the current code in master is badly broken with iterators and I would suggest that we only support lists until we have proper tests for iterators.